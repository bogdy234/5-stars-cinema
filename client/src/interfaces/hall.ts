export interface Hall {
  number: number;
  rows: number;
  columns: number;
}
