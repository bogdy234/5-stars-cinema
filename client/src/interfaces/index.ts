export enum AdminNavOption {
  Movies = "Filme",
  Reservations = "Rezervari",
}
