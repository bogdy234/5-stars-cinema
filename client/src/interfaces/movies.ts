export interface Movie {
    _id: string;
    title: string;
    description: string;
    length: number;
    isRunning: boolean;
    coverImageUrl: string;
    trailerUrl: string;
    rating: number;
    genre: string;
    productionYear: number;
    producer: string;
    direction: string;
    actors: string;
    is3D: boolean;
    isPremiere: boolean;
    runningTimes: { time: Date; hallId: string }[];
}
