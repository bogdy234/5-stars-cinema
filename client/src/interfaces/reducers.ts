import { AdminNavOption } from ".";

export interface AdminReducer {
  selectedOption: AdminNavOption;
}
