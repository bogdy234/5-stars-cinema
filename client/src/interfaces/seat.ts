export enum SeatStatus {
    Unavailable,
    Available,
    Selected,
}

export interface Seat {
    row: number;
    column: number;
    status: SeatStatus;
}