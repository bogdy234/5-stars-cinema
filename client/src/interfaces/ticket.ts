export interface Ticket {
  normal: number;
  reduced: number;
}
