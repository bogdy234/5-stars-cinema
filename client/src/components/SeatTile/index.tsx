import {FC, ReactElement} from 'react';
import {SeatStatus} from "../../interfaces/seat";

interface SeatTileProps{
    seatStatus: SeatStatus;
    seatRow:number;
    seatColumn:number;
}

const SeatTile: FC<SeatTileProps> = ({seatStatus, seatRow, seatColumn}):ReactElement => {
    const getColor = () => {
        if (seatStatus === SeatStatus.Available){
            return 'bg-green-700';
        }
        else if (seatStatus === SeatStatus.Selected){
            return 'bg-red-700';
        }
        else return 'bg-gray-500'
    }

    return (
        <div className={`w-10 h-10 flex justify-center items-center cursor-pointer ${getColor()} ${seatStatus === SeatStatus.Available && 'hover:bg-red-700'}`}>
            {`${seatRow}/${seatColumn}`}
        </div>
    );
};

export default SeatTile;
