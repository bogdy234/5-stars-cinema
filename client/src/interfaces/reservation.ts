import { SeatPosition } from "./seat";
import { Ticket } from "./ticket";

export interface UserReservation {
  _id: string;
  bookedSeats: SeatPosition[];
  movieId: string;
  hallId: string;
  movieTiming: Date;
  ticketsType: Ticket;
  hallNumber: number;
  movieName: string;
  totalPrice: number;
}
