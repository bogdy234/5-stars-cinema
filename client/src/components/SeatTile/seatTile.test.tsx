import renderer from "react-test-renderer";
import { SeatStatus } from "../../interfaces/seat";
import SeatTile from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <SeatTile
        seatStatus={SeatStatus.Available}
        seatRow={0}
        seatColumn={0}
        onPressSeat={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
