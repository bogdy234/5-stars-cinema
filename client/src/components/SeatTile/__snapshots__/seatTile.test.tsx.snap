// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-8 h-8 text-xs sm: text-base sm:w-10 h-10 flex justify-center items-center cursor-pointer bg-green-700 hover:bg-red-700"
  onClick={[MockFunction]}
>
  0/0
</div>
`;
